### ISSUE TITLE
Teleport strange behavior: setting "position" has no effect

### ISSUE THREAD

**Issue by zkytony** [allenai/ai2thor#630]
According to the current documentation for [Teleport](https://ai2thor.allenai.org/robothor/documentation/#sub-teleport), the example code is:
```python
controller.step(
    action="Teleport",
    position=dict(x=0.999, y=1.01, z=-0.3541),
    rotation=dict(x=0, y=90, z=0),
    horizon=30
)
```
However, running the following toy example:
```python
from ai2thor.controller import Controller
controller = Controller(scene="FloorPlan_Train1_1")

# Initial pose
event = controller.step("Pass")
print("initial position:", event.metadata["agent"]["position"])
print("initial rotation:", event.metadata["agent"]["rotation"])

# Teleport (example from documentation)
event = controller.step(
    action="Teleport",
    position=dict(x=0.999, y=1.01, z=-0.3541),
    rotation=dict(x=0, y=90, z=0),
    horizon=30
)
print("position after teleport:", event.metadata["agent"]["position"])
print("rotation after teleport:", event.metadata["agent"]["rotation"])
```
I get:
```
initial position: {'x': 3.0, 'y': 0.9009997248649597, 'z': -1.5}
initial rotation: {'x': -0.0, 'y': 269.9995422363281, 'z': 0.0}
position after teleport: {'x': 3.0, 'y': 0.9009997248649597, 'z': -1.5}
rotation after teleport: {'x': -0.0, 'y': 269.9995422363281, 'z': 0.0}
```
So, the Teleport action is **not applied**.

If I do call Telepor using `x`, `y`, `z` arguments instead of `position`:
```python
# Teleport (example from documentation)
event = controller.step(
    action="Teleport",
    x=0.999, y=1.01, z=-0.3541,
    rotation=dict(x=0, y=90, z=0),
    horizon=30
)
```
I **still don't** observe the effect. I had to call "TeleportFull" instead of "Teleport" as follows:
```python
event = controller.step(
    action="TeleportFull",
    x=0.999, y=1.01, z=-0.3541,
    rotation=dict(x=0, y=90, z=0),
    horizon=30
)
```
Then I observe an effect, but it is **not expected**
```
position after teleport: {'x': 1.25, 'y': 0.9009993672370911, 'z': -0.25}
rotation after teleport: {'x': -0.0, 'y': 90.0, 'z': 0.0}
```
Somehow `x` is set to 1.25 even though the specified value is 0.999. Same issue for `y` and `z`.
EDIT: I realized this is because of the "snap to grid" behavior.  That makes sense. 

Could somebody check if this example works in the documentation?

In short, the documentation seems wrong, and "Teleport" seems to not work properly.

I am using ai2thor 2.7.2

**Comment by mattdeitke** [comment]
Sorry for the documentation being a bit ahead. Teleport is currently only supported with `x`, `y`, and `z` positioning. I am planning to update `Teleport`s definition today with https://github.com/allenai/ai2thor/pull/613.

In short, the way it works in 2.7.2 is that `Teleport` only takes in `x`, `y`, and `z` as required parameters (like it used to):
```python
controller.step("Teleport", x=0.5, y=0.5, z=0.5)
```

But now `TeleportFull` requires all degrees of freedom to be specified. However, the problem with the LoCoBot and Drone agent's is that they do not have the same degrees of freedom that the default agent has. In particular, they do not have the ability to stand. Right now, `TeleportFull` basically requires:
```python
controller.step(
    action="TeleportFull",
    position=dict(x=0, y=0, z=0),
    rotation=dict(x=0, y=90, z=0),
    horizon=30,
    standing=True
)
```
where `standing=True` is required for _every_ agent (even the LoCoBot), which is why `TeleportFull` is not working in your case.

This is going to be fixed today, and match the definitions shown in the documentation.

**Comment by zkytony** [comment]
>  Teleport is currently only supported with x, y, and z positioning. I am planning to update Teleports definition today with #613
Does this mean that the examples I pasted for "Teleport" is invalid because I specified rotation together with x, y, z?

Would be nice if there are separate examples Teleport and TeleportFull, and both involve changing positions and rotations (and copy/paste the example code should work).

Also, I haven't noticed `standing` as an option and not sure what it does. Does this matter to me if I am using LoCoBot? If `standing=True` is required but I didn't specify it, would you throw a warning or something? If it's required why not set it to be True by default?

**Comment by zkytony** [comment]
Also is it possible to maintain documentation for different versions of ai2thor, like what Python does? It appears that the releases are quite frequent (understandably) and sooner or later I'll be reading documentation that's not for the version of ai2thor I'm running. Would be great to refer to documentation of previous versions.

**Comment by mattdeitke** [comment]
> It appears that the releases are quite frequent (understandably) and sooner or later I'll be reading documentation that's not for the version of ai2thor I'm running. Would be great to refer to documentation of previous versions.

This is what I'm working on now, which is one of the reasons the documentation is a tad behind what's possible. I've set up the automated process from the javascript side, but basically just need to write a scraper and add comments to all Unity functions.

**Comment by mattdeitke** [comment]
> Does this mean that the examples I pasted for "Teleport" is invalid because I specified rotation together with x, y, z? (That feels weird. And upon reading "only takes in x, y, and z as required parameters" I think that "Teleport" example should produce the same output as the "TeleportFull" example).

I suspect `Teleport` and `TeleportFull` are probably not even being called in your example, because they currently do not correspond to valid function signatures from the Unity side: (see [here for Teleport](https://github.com/allenai/ai2thor/blob/master/unity/Assets/Scripts/BaseFPSAgentController.cs#L1917) and [here for TeleportFull](https://github.com/allenai/ai2thor/blob/master/unity/Assets/Scripts/BaseFPSAgentController.cs#L1898).

> Would be nice if there are separate examples Teleport and TeleportFull, and both involve changing positions and rotations (and copy/paste the example code should work).

This will be fixed today in https://github.com/allenai/ai2thor/pull/613, which should go into the next release.

> Also, I haven't noticed standing as an option and not sure what it does. Does this matter to me if I am using LoCoBot? If standing=True is required but I didn't specify it, would you throw a warning or something? 

Yeah, `standing` is a current bug that is also being fixed in #613. Basically, the LoCoBot's body has no concept of `standing`. I suspect passing in `standing=True` just doesn't do anything, but I'm not terribly certain and am fixing this.

> If it's required why not set it to be True by default?

It is required for the default iTHOR agent, but not any of the other agents. Which is why the differences arise.

**Comment by mattdeitke** [comment]
With the new release of v2.7.4., Teleport and TeleportFull are now properly documented and fix the mentioned bugs. If there are any related issues, feel free to reopen :)
